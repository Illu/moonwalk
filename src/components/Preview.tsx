import React from "react";
import styled from "styled-components/native";

const ItemWrapper = styled.TouchableOpacity`
  border-radius: 30px;
  background: #000000;
  flex: 1;
  overflow: hidden;
  margin: 20px;
  justify-content: space-between;
`;

const Title = styled.Text`
  color: white;
  font-weight: bold;
  font-size: 20px;
  margin: 20px;
`;

const Time = styled.Text`
  color: white;
  font-weight: bold;
  font-size: 20px;
  margin: 20px;
  align-self: flex-end;
`;

const Thumbnail = styled.ImageBackground`
  height: 100%;
  justify-content: space-between;
`;

interface Props {
  data: any; // TODO: add launch interface
  onPress: () => void;
}

const Preview = ({ data, onPress }: Props) => (
  <ItemWrapper onPress={onPress}>
    <Thumbnail
      source={{ uri: data.image || data.rocket.configuration.image_url }}
      imageStyle={{ opacity: 0.9 }}
    >
      <Title>{data.name}</Title>
      <Time>{new Date(data.net).toLocaleString()}</Time>
    </Thumbnail>
  </ItemWrapper>
);

export default Preview;
