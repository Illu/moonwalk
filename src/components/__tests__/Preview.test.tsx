import React from "react";

import TestRenderer from "../../helpers/testRenderer";
import Preview from "../Preview";

describe("Preview", () => {
  it("renders correctly", () => {
    const tree = TestRenderer(
      <Preview
        data={{ image: "IMAGE_URL", name: "NAME", net: "000000" }}
        onPress={jest.fn}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
