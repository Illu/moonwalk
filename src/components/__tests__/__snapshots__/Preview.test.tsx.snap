// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#000000",
      "borderRadius": 30,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "space-between",
      "marginBottom": 20,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 20,
      "opacity": 1,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "height": "100%",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "IMAGE_URL",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "opacity": 0.9,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
          },
        ]
      }
    >
      NAME
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-end",
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
          },
        ]
      }
    >
      1/1/2000, 12:00:00 AM
    </Text>
  </View>
</View>
`;
